### PR TITLE
chore(test): implement podman extensions handling test

### PR DESCRIPTION
**Pull request by ScrewTSW**
### What does this PR do?
* Implements Podman Extension smoke test
* Modifies package.json to simplify yarn commands, allow build without having to run entire smoke suite

### What issues does this PR fix or reference?
#3789 

### How to test this PR?

* append
```
"test:e2e:smoke:run:podman-extension": "xvfb-maybe vitest run tests/src/podman-extension-smoke.spec.ts --threads false",
```
into `package.json`

* run `yarn run test:e2e:build`
* run `yarn run test:e2e:smoke:run:podman-extension`

### Example test execution
```
$ yarn run test:e2e:smoke:run:settings-extensions
yarn run v1.22.19
$ xvfb-maybe vitest run tests/src/settings-extensions-smoke.spec.ts --threads false
Cleaning up folder: tests/output
Folder found, removing...

 RUN  v0.34.6 /home/tdancs/Workspace/podman-desktop

stdout | unknown test
video will be written to: tests/output/videos
podman desktop custom config will be written to: tests/output/podman-desktop

stdout | unknown test
STDERR: [compose] Error getting compose version: Error: Failed to execute command: spawn /home/tdancs/Workspace/podman-desktop/tests/output/podman-desktop/extensions-storage/podman-desktop.compose/bin/docker-compose ENOENT

main ↪️ Activating extension (podman-desktop.compose) ended in 142 milliseconds
main ↪️ [kind] kind extension is active

stdout | unknown test
main ↪️ Activating extension (podman-desktop.kind) ended in 77 milliseconds
main ↪️ [compose] Error getting compose version: Error: Failed to execute command: spawn /home/tdancs/Workspace/podman-desktop/tests/output/podman-desktop/extensions-storage/podman-desktop.compose/bin/docker-compose ENOENT

stdout | unknown test
main ↪️ Activating extension (podman-desktop.kube-context) ended in 82 milliseconds

stdout | unknown test
MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 extensions-started listeners added. Use emitter.setMaxListeners() to increase limit
    at _addListener (node:electron/js2c/sandbox_bundle:2:32039)
    at EventEmitter.addListener (node:electron/js2c/sandbox_bundle:2:34927)
    at Object.receive (<anonymous>:31:20)

stdout | unknown test
main ↪️ [podman] Error while pinging docker as podman Error: connect ENOENT /var/run/docker.sock
    at PipeConnectWrap.afterConnect [as oncomplete] (node:net:1495:16)
main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 301.0193649996072ms
main ↪️ Activating extension (podman-desktop.registries) ended in 109 milliseconds

stdout | unknown test
main ↪️ Activating extension (podman-desktop.podman) ended in 221 milliseconds
main ↪️ PluginSystem: initialization done.
main ↪️ Development mode, skipping startup install

stdout | unknown test
main ↪️ PluginSystem: received dom-ready event from the UI
main ↪️ Delayed startup, flushing 52 events

stdout | unknown test
main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 211.67385899834335ms

stdout | unknown test
main ↪️ [kind] kind extension is active
main ↪️ [kind] kind extension is active
main ↪️ [kind] kind extension is active
main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 166.46360299922526ms
main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 159.8896709997207ms

stdout | unknown test
%cElectron Security Warning (Disabled webSecurity) font-weight: bold; This renderer process has "webSecurity" disabled. This
  exposes users of this app to severe security risks.

For more information and help, consult
https://electronjs.org/docs/tutorial/security.
This warning will not show up
once the app is packaged.
%cElectron Security Warning (allowRunningInsecureContent) font-weight: bold; This renderer process has "allowRunningInsecureContent"
  enabled. This exposes users of this app to severe security risks.

  
For more information and help, consult
https://electronjs.org/docs/tutorial/security.
This warning will not show up
once the app is packaged.
%cElectron Security Warning (Insecure Content-Security-Policy) font-weight: bold; This renderer process has either no Content Security
  Policy set or a policy with "unsafe-eval" enabled. This exposes users of
  this app to unnecessary security risks.

For more information and help, consult
https://electronjs.org/docs/tutorial/security.
This warning will not show up
once the app is packaged.

stdout | unknown test
Telemetry enablement: false

stdout | tests/src/settings-extensions-smoke.spec.ts > Basic e2e verification of podman desktop settings-extensions > Verify podman is running and available > Disable Podman extension and verify status
main ↪️ [podman] stopping podman extension

stdout | tests/src/settings-extensions-smoke.spec.ts > Basic e2e verification of podman desktop settings-extensions > Verify podman is running and available > Disable Podman extension and verify status
main ↪️ [podman] No machine to stop

stdout | tests/src/settings-extensions-smoke.spec.ts > Basic e2e verification of podman desktop settings-extensions > Verify podman is running and available > Re-Enable Podman extension from Settings -> Extensions page and verify podman status
STDERR: Font podman-desktop.podman-podman.woff2 already registered.


stdout | tests/src/settings-extensions-smoke.spec.ts > Basic e2e verification of podman desktop settings-extensions > Verify podman is running and available > Re-Enable Podman extension from Settings -> Extensions page and verify podman status
STDERR: Icon podman-icon already registered.


stdout | tests/src/settings-extensions-smoke.spec.ts > Basic e2e verification of podman desktop settings-extensions > Verify podman is running and available > Re-Enable Podman extension from Settings -> Extensions page and verify podman status
main ↪️ Font podman-desktop.podman-podman.woff2 already registered.
main ↪️ Icon podman-icon already registered.

stdout | tests/src/settings-extensions-smoke.spec.ts > Basic e2e verification of podman desktop settings-extensions > Verify podman is running and available > Re-Enable Podman extension from Settings -> Extensions page and verify podman status
main ↪️ Activating extension (podman-desktop.podman) with max activation time of 10 seconds

stdout | tests/src/settings-extensions-smoke.spec.ts > Basic e2e verification of podman desktop settings-extensions > Verify podman is running and available > Re-Enable Podman extension from Settings -> Extensions page and verify podman status
main ↪️ [podman] Error while pinging docker as podman Error: connect ENOENT /var/run/docker.sock
    at PipeConnectWrap.afterConnect [as oncomplete] (node:net:1495:16)
main ↪️ [kind] kind extension is active

stdout | tests/src/settings-extensions-smoke.spec.ts > Basic e2e verification of podman desktop settings-extensions > Verify podman is running and available > Re-Enable Podman extension from Settings -> Extensions page and verify podman status
main ↪️ Activating extension (podman-desktop.podman) ended in 215 milliseconds

stdout | unknown test
STDERR: Waiting for the debugger to disconnect...


stdout | unknown test
Elapsed time of closing the electron app: 111.08883500099182 ms

stdout | unknown test
Saving a video file took: 4.706040000542998 ms
Video file saved as: tests/output/videos/settings-extensions-e2e.webm

 ✓ tests/src/settings-extensions-smoke.spec.ts (5) 4601ms
   ✓ Basic e2e verification of podman desktop settings-extensions (5) 2284ms
     ✓ Verify podman is running and available (5) 2284ms
       ✓ Check Podman present on Dashboard page 333ms
       ✓ Check Podman is running 352ms
       ✓ Open on Podman extension tab
       ✓ Disable Podman extension and verify status 351ms
       ✓ Re-Enable Podman extension from Settings -> Extensions page and verify podman status 1088ms

 Test Files  1 passed (1)
      Tests  5 passed (5)
   Start at  12:46:45
   Duration  6.29s (transform 467ms, setup 25ms, collect 1.08s, tests 4.60s, environment 317ms, prepare 83ms)

Done in 7.10s.
   ```

### CODE REVIEW

**Review by benoitf — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,4 @@ dist
 **/coverage
 # kind
 kind.exe
+**/untracked*
```
Comment (odockal) [inline .gitignore]: What is this change for?

File: tests/src/podman-extension-smoke.spec.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    : await playExpect(podmanProviderLocator).not.toBeVisible();
+  // always present and visible
+  const settingsExtensionsPage = await openSettingsExtensionsPage();
+  const settingsExtensionsPodmanLocator = settingsExtensionsPage.getFeaturedExtension('Podman');
+  await playExpect(settingsExtensionsPodmanLocator).toBeVisible();
+  const podmanExtensionRowLocator = settingsExtensionsPage.getExtensionRowFromTable('Podman (default extension)');
+  await playExpect(podmanExtensionRowLocator).toBeVisible();
+  const connectionStatusLabel = podmanExtensionRowLocator.getByLabel('connection-status-label');
+  await playExpect(connectionStatusLabel).toBeVisible();
+  const connectionStatusLocatorText = await connectionStatusLabel.innerText();
```
Comment (odockal) [inline tests/src/podman-extension-smoke.spec.ts]: Please add a timeout in the options object. default is quite long. 2/3sec perhaps.

File: tests/src/podman-extension-smoke.spec.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    await playExpect(podmanExtensionPage.status.getByText('ENABLED')).toBeVisible();
+  } else {
+    await playExpect(podmanExtensionPage.enableButton).toBeEnabled({ timeout: 10000 });
+    await playExpect(podmanExtensionPage.disableButton).toBeDisabled({ timeout: 10000 });
+    await playExpect(podmanExtensionPage.status.getByText('DISABLED')).toBeVisible();
+  }
+  // expand Settings -> Preferences menu
+  await settingsBar.preferencesTab.click();
+  enabled
+    ? await playExpect(settingsBar.getSettingsNavBarTabLocator('Extension: Podman')).toBeVisible()
```
Comment (odockal) [inline tests/src/podman-extension-smoke.spec.ts]: please Extract strings into test file variables/constants.

**Review by odockal — CHANGES_REQUESTED** [review]

File: tests/src/podman-extension-smoke.spec.ts (hunk 2 above)
Comment (ScrewTSW) [reply tests/src/podman-extension-smoke.spec.ts]: fixed

File: tests/src/podman-extension-smoke.spec.ts (hunk 3 above)
Comment (ScrewTSW) [reply tests/src/podman-extension-smoke.spec.ts]: fixed.

File: .gitignore (hunk 1 above)
Comment (ScrewTSW) [reply .gitignore]: to be able to keep WIP files between sources, without git suggesting them for stage, they keep getting added when I just do git add test/.
It's not necessary, but is it harmful?

**Review by ScrewTSW — COMMENTED** [review]
Fixed review points, responded

File: .gitignore (hunk 1 above)
Comment (odockal) [reply .gitignore]: Might not be harmful, but totally not expected. I would say try to get it in as a part of individual PR so it is clear such change was introduced.

Comment (ScrewTSW) [reply .gitignore]: ACK

**Review by odockal — APPROVED** [review]

**Pull request merged**